components/ping_ai_copilot/extension/content/content_scripts/content.js

```diff
@@ -0,0 +1,156 @@
+import "../ui/style.css"
+let summarizer = null;
+let summaryBox = null;
+let isTextSelected = false;
+
+const initializeExtension = () => {
+  document.removeEventListener('mouseup', handleTextSelection);
+  document.addEventListener('mouseup', handleTextSelection);
+  document.addEventListener('click', handleDocumentClick);
+}
+
+const handleTextSelection = (event) => {
+  const selection = window.getSelection();
+  const selectedText = selection.toString().trim();
+  if (selectedText.length > 0 && !(summaryBox && summaryBox.contains(selection.anchorNode))) {
+    isTextSelected = true
+    showSummarizeIcon(selectedText, event);
+  } else {
+    hideSummarizeIcon();
+  }
+}
+
+const handleDocumentClick = (event) => {
+  if (
+    summaryBox &&
+    !summaryBox.contains(event.target) &&
+    (!summarizer || !summarizer.contains(event.target))
+  ) {
+    hideSummaryBox();
+  }
+}
+
+const showSummarizeIcon = (selectedText, event) => {
+  if (!summarizer) {
+    summarizer = document.createElement('div');
+    summarizer.id = 'summarizer-icon';
+    
+    const iconImage = document.createElement('img');
+    iconImage.src = chrome.runtime.getURL('extension/assets/aiSummarizerIcon.svg');
+    iconImage.alt = 'Summarize Icon';
+    iconImage.id = 'iconImage';
+    
+    summarizer.appendChild(iconImage);
+
+    summarizer.addEventListener('click', (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      sendTextToSummarize(selectedText);
+    });
+    document.body.appendChild(summarizer);
+  }
+
+  if (isTextSelected) {
+    summarizer.style.top = `${event.clientY}px`;
+    summarizer.style.left = `${event.clientX}px`;
+  }
+
+  summarizer.style.display = 'inline-block';
+}
+
+const hideSummarizeIcon = () => {
+  if (summarizer) {
+    summarizer.style.display = 'none';
+  }
+}
+
+const sendTextToSummarize = (text) => {
+  showSummaryBox(true);
+  try {
+    chrome.runtime.sendMessage({ action: 'summarize', text: text });
+  } catch (error) {
+    console.error('Failed to send message:', error);
+    showSummaryBox(false, 'An error occurred. Please refresh the page and try again.');
```

Comment:
will the user need to refresh. or user can click somewhere else or select again to trigger this again?